sts-mock/src/token/tokenEncrypter/tokenEncrypter.ts

```diff
@@ -0,0 +1,106 @@
+import { errorResult, Result, successResult } from "../../utils/result";
+import { createPublicKey, JsonWebKey, KeyObject } from "node:crypto";
+import { CompactEncrypt } from "jose";
+import { JWT } from "../tokenSigner/tokenSigner";
+
+export type JWE = `${string}.${string}.${string}.${string}.${string}`;
+
+export class TokenEncrypter implements ITokenEncrypter {
+  private readonly jwksUri: string;
+
+  constructor(jwksUri: string) {
+    this.jwksUri = jwksUri;
+  }
+
+  async encrypt(jwt: JWT): Promise<Result<JWE>> {
+    const getJwksResult = await this.getJwks();
+    if (getJwksResult.isError) {
+      return getJwksResult;
+    }
+
+    const extractEncryptionKeyResult = this.getEncryptionKey(
+      getJwksResult.value,
+    );
+    if (extractEncryptionKeyResult.isError) {
+      return extractEncryptionKeyResult;
+    }
+    const encryptionKey = extractEncryptionKeyResult.value;
+
+    const header = {
+      alg: "RSA-OAEP-256",
+      enc: "A256GCM",
+    };
+
+    try {
+      const jwe = (await new CompactEncrypt(new TextEncoder().encode(jwt))
+        .setProtectedHeader(header)
+        .encrypt(encryptionKey)) as JWE;
+      return successResult(jwe);
+    } catch {
+      return errorResult({
+        errorMessage: "Error encrypting token",
+        errorCategory: "SERVER_ERROR",
+      });
+    }
+  }
+
+  private async getJwks(): Promise<Result<object>> {
+    let response;
+    try {
+      response = await fetch(this.jwksUri, { method: "GET" });
+      if (!response.ok) {
+        return errorResult({
+          errorMessage: "Error fetching JWKS",
+          errorCategory: "SERVER_ERROR",
+        });
+      }
+    } catch {
+      return errorResult({
+        errorMessage: "Unexpected network error fetching JWKS",
+        errorCategory: "SERVER_ERROR",
+      });
+    }
+
+    let body: object;
+    try {
+      body = JSON.parse(await response.text());
```

Comment:
```suggestion
      body = await response.json();
```

Reply:
We found this on Friday when we were making a similar func. Not better just thought I'd share it, both ways work just fine 😄

Reply:
Yes, I did use `response.json()` initially, but I switched back to `JSON.parse()` because I am more familiar with it, but I can change it.

Reply:
Updated.